02stl/cpp/particlesMap.cc

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <map>
+#include <string>
+#include <iostream>
+
+/// func
+void keepMax(std::map<std::string,float>& theMap, const std::string key, const float value)
+{
+  if (theMap.find(key)==theMap.end()) //element doesn't already exist in map
+    theMap[key] = value;
+  else
+    // any logic can go in here, eg a counter, an average, etc...
+    theMap[key] = std::max(theMap[key],value);
+}
+
+/// class
+class maxMap
+{
+public:
+  typedef std::map<std::string,float>::iterator iterator;
+  
+  void insert(const std::pair<std::string,float> theElement)
+  {
+    std::string key = theElement.first;
+    float value = theElement.second;
+    if (m_map.find(key)==m_map.end()) //element doesn't already exist in map
+      m_map[key] = value;
+    else
+      // any logic can go in here, eg a counter, an average, etc...
+      m_map[key] = std::max(m_map[key],value);
+  }
+  iterator begin() { return m_map.begin(); }
+  iterator end() { return m_map.end(); }
+private:
+  std::map<std::string,float> m_map;
+};
+
+
+///main
+int main()
+{
+  FILE* ifp = fopen("02stl/cpp/particleList.txt","r");
+
+  // Read in the data
+  maxMap theParticles;
+  char name[80];
+  float momentum;
```

Comment:
On 64bit systems, I would rather they learn to use `double` rather than `float`.